### PR TITLE
[Resource] Add deployment target Argument

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPIInstanceDeploymentTarget
--- PASS: TestAccIBMPIInstanceDeploymentTarget (1049.11s)
PASS

...
```

### CODE REVIEW

File: website/docs/r/pi_instance.html.markdown [hunk 1]
```diff
@@ -64,6 +64,12 @@ Review the argument references that you can specify for your resource.
 - `pi_anti_affinity_instances` - (Optional, String) List of pvmInstances to base storage anti-affinity policy against; required if requesting `anti-affinity` and `pi_anti_affinity_volumes` is not provided.
 - `pi_anti_affinity_volumes`- (Optional, String) List of volumes to base storage anti-affinity policy against; required if requesting `anti-affinity` and `pi_anti_affinity_instances` is not provided.
 - `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
+- `pi_deployment_target` - (Optional, List) The deployment of a dedicated host. MaxItems:1.
+  
+  Nested scheme for `pi_deployment_target` :
+  * `id` - (Required, String) The uuid of the host group or host.
+  * `type` - (Required, String) The deployment target type.Supported values are `host` and `hostGroup`.
```
Comment (ismirlia) [inline website/docs/r/pi_instance.html.markdown:71]:
```suggestion
  * `type` - (Required, String) The deployment target type. Supported values are `host` and `hostGroup`.
```

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 2]
```diff
@@ -146,6 +147,27 @@ func ResourceIBMPIInstance() *schema.Resource {
...
+				Description: "The deployment of a dedicated host",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_ID: {
+							Description: "The uuid of the host group or host",
+							Required:    true,
+							Type:        schema.TypeString,
+						},
+						Attr_Type: {
+							Description:  "specify if deploying to a host group or a host",
```
Comment (ismirlia) [inline ibm/service/power/resource_ibm_pi_instance.go:160]: Have description here match the description in the markdown?
```suggestion
							Description:  "The deployment target type. Supported values are 'host' and 'hostGroup'.",
```

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 3]
```diff
@@ -146,6 +147,27 @@ func ResourceIBMPIInstance() *schema.Resource {
 				Default:     true,
 				Description: "Indicates if all volumes attached to the server must reside in the same storage pool",
 			},
+			Arg_DeploymentTarget: {
+				Description: "The deployment of a dedicated host",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_ID: {
+							Description: "The uuid of the host group or host",
```
Comment (ismirlia) [inline ibm/service/power/resource_ibm_pi_instance.go:155]:
```suggestion
							Description: "The uuid of the host group or host.",
```

File: website/docs/r/pi_instance.html.markdown [hunk 4]
```diff
@@ -64,6 +64,12 @@ Review the argument references that you can specify for your resource.
 - `pi_anti_affinity_instances` - (Optional, String) List of pvmInstances to base storage anti-affinity policy against; required if requesting `anti-affinity` and `pi_anti_affinity_volumes` is not provided.
 - `pi_anti_affinity_volumes`- (Optional, String) List of volumes to base storage anti-affinity policy against; required if requesting `anti-affinity` and `pi_anti_affinity_instances` is not provided.
 - `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
+- `pi_deployment_target` - (Optional, List) The deployment of a dedicated host. MaxItems:1.
```
Comment (ismirlia) [inline website/docs/r/pi_instance.html.markdown:67]: Is Max Items here's the maximum length of the list? As in you can at most specify one target?
```suggestion
- `pi_deployment_target` - (Optional, List) The deployment of a dedicated host. Max Items: 1.
```

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 5]
```diff
@@ -146,6 +147,27 @@ func ResourceIBMPIInstance() *schema.Resource {
 				Default:     true,
 				Description: "Indicates if all volumes attached to the server must reside in the same storage pool",
 			},
+			Arg_DeploymentTarget: {
+				Description: "The deployment of a dedicated host",
```
Comment (ismirlia) [inline ibm/service/power/resource_ibm_pi_instance.go:151]:
```suggestion
				Description: "The deployment of a dedicated host. Max Items: 1.",
```

Comment (michaelkad) [reply ibm/service/power/resource_ibm_pi_instance.go:152]: That is not needed since `MaxItems` is define here.

**Review by ismirlia — APPROVED** [review]

**Comment by michaelkad** [comment]
Merged

**Pull request closed without merging**